src/consensus/proposer.rs

```diff
@@ -230,15 +267,14 @@ impl BlockProposer {
             let timeout = time::sleep_until(deadline);
             select! {
                 _ = poll_interval.tick() => {
-                    if let Ok(decision) = self.shard_decision_rx.try_recv() {
-                       if let Some(proto::full_proposal::ProposedValue::Shard(chunk)) = decision.proposed_value {
-                            let chunk_height = chunk.header.clone().unwrap().height.unwrap();
-                            let chunk_block_number = chunk_height.block_number;
-                            if self.pending_chunks.contains_key(&chunk_block_number) {
-                                self.pending_chunks.get_mut(&chunk_block_number).unwrap().push(chunk);
-                            } else {
-                                self.pending_chunks.insert(chunk_block_number, vec![chunk]);
-                            }
+                    // TODO(aditi): This breaks if syncd shard chunks show up in shard_decision_rx.
+                    if let Ok(chunk) = self.shard_decision_rx.try_recv() {
```

Comment:
This makes me think we should separate sync and consensus channels. We might want to start up consensus, buffer all the messages while sync completes and then process them. No need to change anything right now, but something we'll need to consider how to handle.